src/java_bytecode/java_object_factory.cpp

```diff
@@ -571,9 +571,15 @@ codet initialize_nondet_string_struct(
 
   // `obj` is `*expr`
   const struct_typet &struct_type = to_struct_type(ns.follow(obj.type()));
-  const irep_idt &class_id = "java::" + id2string(struct_type.get_tag());
-
-  // @clsid = String and @lock = false:
+  // @clsid = java::java.lang.String or similar
+  // Note that CharSequence cannot be used as classid because it's abstract
+  // \todo allow StringBuffer and StringBuilder as classid for Charsequence
+  const irep_idt &class_id =
```

Comment:
Explain in the comment why you're allowing classes other than `String` to be aliased with `String` here?